packaging/installer/netdata-uninstaller.sh

```diff
@@ -55,15 +58,58 @@ if [ "$YES" != "1" ]; then
   exit 1
 fi
 
-if [[ $EUID -ne 0 ]]; then
+if [ "$(id -u)" -ne 0 ]; then
   echo >&2 "This script SHOULD be run as root or otherwise it won't delete all installed components."
   key="n"
-  read -r -s -n 1 -p "Do you want to continue as non-root user [y/n] ? " key
+  read -r 1 -p "Do you want to continue as non-root user [y/n] ? " key
   if [ "$key" != "y" ] && [ "$key" != "Y" ]; then
     exit 1
   fi
 fi
 
+user_input() {
+  TEXT="$1 [y/n]"
+
+  while true; do
+    echo "$TEXT"
+    read -r yn
+
+    case "$yn" in
+       [Yy]*) return 0;;
+       [Nn]*) return 1;;
+       *) echo "Please answer yes or no.";;
+     esac
+   done
+}
+
+if [ -x "$(command -v apt-get)" ]; then
+  if dpkg -s netdata > /dev/null; then
+    echo "Found netdata native installation"
+	if user_input "Do you want to remove netdata? "; then
+      apt-get remove netdata ${FLAG}
+	fi
+    if dpkg -s netdata-repo-edge > /dev/null; then
+	  if user_input "Do you want to remove netdata-repo-edge? "; then
+        apt-get remove netdata-repo-edge ${FLAG}
+      fi
+    fi
+    exit 0
+  fi
+elif [ -x "$(command -v dnf)" ]; then
+  if rpm -q netdata > /dev/null; then
+    echo "Found netdata native installation."
+	if user_input "Do you want to remove netdata? "; then
+      dnf remove netdata ${FLAG}
+	fi
+    if rpm -q netdata-repo-edge > /dev/null; then
+	  if user_input "Do you want to remove netdata-repo-edge? "; then
+        dnf remove netdata-repo-edge ${FLAG}
+	  fi
+    fi
+    exit 0
+  fi
+fi
```

Comment:
I see four issues here:

1. This does not handling the case of a stable install, which uses `netdata-repo` and not `netdata-repo-edge`.
2. This code is not handling interactivity correctly. In some cases, users may get prompted twice (once by the script and once by the package manager), while a non-interactive run is not possible because of how `user_input` works.
3. This is missing handling for CentOS 7, which we do provide packages for. The code there should be essentially the same as for DNF, but with YUM instead.
4. This is missing handling for openSUSE, which we also provide packages for. That particular case should probably use Zypper.

Reply:
Also, indentation is strange and inconsistent in a number of places in this block.